### ISSUE TITLE
Improve documentation on basic stuff

### ISSUE THREAD

**Issue by karakostis** [reactstrap/reactstrap#585]
Great library. Even greater that supports Bootstrap4. 
But the documentation is really poor. Its not clear to me how to make the most basic thing needed in bootstrap. 
That is: Create a full-height, auto-resize columns template. Yes, sure there is all this part of documentation with the Grid and the different props you can pass in. And yes we have the flexbox in Bootstrap4 which allows to have auto-resize columns but how do you do that with reactstrap? 
Building the skeleton (template) of the page is the first thing but still there are no examples!

Some more examples would be helpful.

**Comment by TheSharpieOne** [comment]
I understand and this is something we are wanting to address. I know there is a WIP PR to change the docs to gatsby and others have asked when adding docs in PRs if it should include all of the examples bootstrap have (and the answer is yes, even if the examples have nothing special for reactstrap)
Right now the docs are mostly documenting the props which are available and a developer would need to reference bootstrap for possible values to pass to those props. For the `Col` component the [reactstrap documentation](https://reactstrap.github.io/components/layout/) has an example with `auto` and does specify values for `Col`'s props. While it is mostly "code" explanation/documentation of the props and values, it says to supply `"auto"` when you want a breakpoint to be auto sized.

```
const stringOrNumberProp = PropTypes.oneOfType([PropTypes.number, PropTypes.string]);
const columnProps = PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.number,
  PropTypes.bool,
  PropTypes.shape({
    size: PropTypes.oneOfType([PropTypes.bool, PropTypes.number, PropTypes.string]),
    // example size values:
    // 12 || "12" => col-12 or col-`width`-12
    // auto => col-auto or col-`width`-auto
    // true => col or col-`width`
    push: stringOrNumberProp,
    pull: stringOrNumberProp,
    offset: stringOrNumberProp
  })
]);

Col.propTypes = {
  xs: columnProps,
  sm: columnProps,
  md: columnProps,
  lg: columnProps,
  xl: columnProps,
  // override the predefined width (the ones above) with your own custom widths.
  // see https://github.com/reactstrap/reactstrap/issues/297#issuecomment-273556116
  widths: PropTypes.array,
}
```

**Comment by aarongodin** [comment]
I'm curious on the state of this, because to echo the original post this is my favorite bootstrap/react library, but I want it to have amazing documentation. Gatsby looks like a great solution!

On that note, the docs would benefit from a lot more description. Having so many examples is great, but contextualizing those with some explanation would be great.

If moving to gatsby is something you're still into, I would love to contribute.

**Comment by TheSharpieOne** [comment]
Yes, Gatsby would be awesome! It's a lot of work just to migrate. Enhancing can come with that effort or shortly afterwards.
We really need a better standard for what we document and how we document it. For example, some components have their own page while others are group under a single page. It was supposed to mimic bootstrap's grouping/structure but somewhere it fell apart.
I would love to migrate to Gatsby and almost mimic bootstrap's docs 1-to-1 (example-to-example; where applicable) and add the special reactstrap sauce on top (such as `tag`, `cssModules`, and other utilities).
I know another pain point is "what are the acceptable values for these props" and I always point out that "any value can be used since bootstrap can be extended", but maybe we should document the default values bootstrap allows out of the box and explain how to create custom variants by extending bootstrap (and I am fine with linking to bootstrap's docs when it makes sense).

**Comment by glrodasz** [comment]
What about using a project like https://www.docz.site/? This project looks great, but I have to jump between the poor documentation and the bootstrap original docs, and sometimes just guess what are the possible values to reactstrap.
